src/test-jdk14/java/com/fasterxml/jackson/databind/records/RecordImplicitSingleValueUsePropertiesBasedCreatorsTest.java

```diff
@@ -0,0 +1,75 @@
+{
+
+    private final ObjectMapper MAPPER = jsonMapperBuilder()
+            .annotationIntrospector(new Jdk8ConstructorParameterNameAnnotationIntrospector())
+            .constructorDetector(ConstructorDetector.USE_PROPERTIES_BASED)
+            .build();
+
+    record RecordWithMultiValueCanonAndSingleValueAltConstructor(int id, String name) {
+
+        public RecordWithMultiValueCanonAndSingleValueAltConstructor(int id) {
+            this(id, "AltConstructor");
+        }
+    }
+
+    record RecordWithSingleValueCanonAndMultiValueAltConstructor(String name) {
+
+        public RecordWithSingleValueCanonAndMultiValueAltConstructor(String name, String email) {
+            this("AltConstructor");
+        }
+    }
+
+    /*
+    /**********************************************************************
+    /* Test methods, multi-value canonical constructor + single-value alt constructor
+    /**********************************************************************
+     */
+
+    public void testDeserializeMultipleConstructors_UsingMultiValueCanonicalConstructor() throws Exception {
+        RecordWithMultiValueCanonAndSingleValueAltConstructor value = MAPPER.readValue(
+                a2q("{'id':123,'name':'Bob'}"),
+                RecordWithMultiValueCanonAndSingleValueAltConstructor.class);
+
+        assertEquals(new RecordWithMultiValueCanonAndSingleValueAltConstructor(123, "Bob"), value);
+    }
+
+    public void testDeserializeMultipleConstructors_WillIgnoreSingleValueAltConstructor() throws Exception {
+        RecordWithMultiValueCanonAndSingleValueAltConstructor value = MAPPER.readValue(
+                a2q("{'id':123}"),
+                RecordWithMultiValueCanonAndSingleValueAltConstructor.class);
+
+        assertEquals(new RecordWithMultiValueCanonAndSingleValueAltConstructor(123, null), value);
+    }
+
+    /*
+    /**********************************************************************
+    /* Test methods, single-value canonical constructor + multi-value alt constructor
+    /**********************************************************************
+     */
+
+    public void testDeserializeMultipleConstructors_UsingSingleValueCanonicalConstructor() throws Exception {
+        RecordWithSingleValueCanonAndMultiValueAltConstructor value = MAPPER.readValue(
+                a2q("{'name':'Bob'}"),
+                RecordWithSingleValueCanonAndMultiValueAltConstructor.class);
+
+        assertEquals(new RecordWithSingleValueCanonAndMultiValueAltConstructor("Bob"), value);
+    }
+
+    public void testDeserializeMultipleConstructors_WillIgnoreMultiValueAltConstructor() throws Exception {
+        try {
+            MAPPER.readValue(a2q("{'name':'Bob','email':'bob@email.com'}"), RecordWithSingleValueCanonAndMultiValueAltConstructor.class);
+        } catch (UnrecognizedPropertyException e) {
+            verifyException(e, "Unrecognized");
+            verifyException(e, "\"email\"");
```

Comment:
It's different because even this will match: `verifyException(e, "Unrecognized", "this text does not exist in the exception message");`.

`matches` should've been named `anyMatches` and documented with:
> will pass if ANY of the given text matches the exception matches, will fail if NONE of them matches"

Reply:
FWTW I changed the parameter name for 2.16. And yes, it probably makes sense to start using something like AssertJ with regexp matches or such.